### PR TITLE
odf-to-odf: initialize map when adding labels

### PR DESCRIPTION
**Pull request by subhamkrai**
In case of upgrade, we are getting error `panic: assignment to entry in nil map`
because while blockpool creation in 4.10, it was created without labels
and in 4.11 we are trying to assign label which is empty.
This commit fix above issue.

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 1]
```diff
@@ -625,7 +625,15 @@ func (r *StorageClassClaimReconciler) reconcileCephBlockPool() error {
 			return fmt.Errorf("Could not find  device set named default in Storage cluster")
 		}
 		pgUnitSize := util.GetPGBaseUnitSize(deviceSet.Count)
-		r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
+		if r.cephBlockPool.Labels == nil {
+			r.cephBlockPool.Labels = map[string]string{
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+			}
+
+		} else {
```
Comment (Madhu-1) [inline controllers/storageclassclaim/storageclassclaim_controller.go:633]: Remove this one duplicate line https://github.com/red-hat-storage/ocs-operator/blob/c1d20824e55fb233e2dc077d4ebbda60a07582e0/controllers/storageclassclaim/storageclassclaim_controller.go#L461-L463

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 2]
```diff
@@ -625,7 +625,15 @@ func (r *StorageClassClaimReconciler) reconcileCephBlockPool() error {
 			return fmt.Errorf("Could not find  device set named default in Storage cluster")
 		}
 		pgUnitSize := util.GetPGBaseUnitSize(deviceSet.Count)
-		r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
+		if r.cephBlockPool.Labels == nil {
+			r.cephBlockPool.Labels = map[string]string{
```
Comment (Madhu-1) [inline controllers/storageclassclaim/storageclassclaim_controller.go:629]: Make use of `r.cephBlockPool.SetLabels` helper

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 3]
```diff
@@ -625,7 +621,16 @@ func (r *StorageClassClaimReconciler) reconcileCephBlockPool() error {
...
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+			}
+
+		} else {
+			r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
+		}
+		r.cephBlockPool.SetLabels(r.cephBlockPool.Labels)
```
Comment (Madhu-1) [inline controllers/storageclassclaim/storageclassclaim_controller.go:632]:
```suggestion
if r.cephBlockPool.Labels == nil {
			r.cephBlockPool.SetLabels(map[string]string{
				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
			}
			)

		} else {
			r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
		}
```

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 4]
```diff
@@ -625,7 +621,14 @@ func (r *StorageClassClaimReconciler) reconcileCephBlockPool() error {
...
 		}
 		pgUnitSize := util.GetPGBaseUnitSize(deviceSet.Count)
-		r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
+		if r.cephBlockPool.Labels == nil {
+			r.cephBlockPool.SetLabels(map[string]string{
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+			})
+		} else {
+			r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
+		}
```
Comment (nb-ohad) [inline controllers/storageclassclaim/storageclassclaim_controller.go:630]:
```suggestion
		labels := cephBlockPool.GetLabels();
		if labels == nil {
			cephBlockPool.SetLabels(map[string]string{})
		}
		labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
```

or even better, you can do what we do in the deployer, have a util function like this one:
https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/utils/utils.go#L46-L54

then replace the entire section with 
```suggestion
		AddLabel(cephBlockPool, controllers.StorageConsumerNameLabel, r.storageConsumer.Name)
```
And start using it in all places we need to set a label (but don't do it in this PR :) )

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Pull request merged**